lib/db/mem.js

```diff
@@ -817,6 +819,46 @@ module.exports = function (log, error) {
     return P.resolve({})
   }
 
+  Memory.prototype.createSecurityEvent = function (data) {
+    var addr = data.ipAddr
+    if (ip.isV4Format(addr)) {
+      addr = '::' + addr
+    }
+
+    var event = {
+      createdAt: Date.now(),
+      ipAddr: addr,
+      name: data.name,
+      uid: data.uid,
+      tokenId: data.tokenId
+    }
+    var key = event.uid.toString('hex')
+
+    var events = securityEvents[key] || (securityEvents[key] = [])
+    events.push(event)
+
+    return P.resolve({})
+  }
+
+  Memory.prototype.securityEvents = function (where) {
+    var key = where.uid.toString('hex')
+    var events = securityEvents[key] || []
+    var addr = where.ipAddr
+    if (ip.isV4Format(addr)) {
+      addr = '::' + addr
+    }
+
+    return P.resolve(events.filter(function (ev) {
+      return ev.uid.toString('hex') === key && ip.isEqual(ev.ipAddr, addr)
+    }).map(function (ev) {
+      return {
+        name: ev.name,
+        createdAt: ev.createdAt,
+        verified: (ev.tokenId && !!unverifiedTokens[ev.tokenId.toString('hex')])
```

Comment:
We decided not to take this "join onto the unverifiedTokens table at runtime" approach in the MySQL backend, because it fails when the sessionToken gets deleted.  Does it have the same problem in the mem backend?